### PR TITLE
Support ActionResult changes

### PR DESCRIPTION
**Pull request by alchaussee**
Clad changes made by https://github.com/anki/cozmo-one/pull/2881
Can search for "clad/src" on the "Files changed" tab to see what changed

### CODE REVIEW

File: src/cozmo/action.py [hunk 1]
```diff
@@ -219,9 +210,21 @@ def _recv_msg_robot_completed_action(self, evt, *, msg):
         elif result == types.INTERRUPTED:
             self._set_failed('interrupted', 'Action was interrupted')
 
+        # Can add additional if statements for other ActionResults to exposure more specific failure reasons to users
```
Comment (MarkWesleyAtAnki) [inline src/cozmo/action.py]: typo: exposure should be expose. We also try and wrap at about 80 chars for the SDK. I think we should just remove this comment and add some more detail in the else block though (about how we're summarizing the types by their category).

File: src/cozmo/action.py [hunk 2]
```diff
@@ -219,9 +210,21 @@ def _recv_msg_robot_completed_action(self, evt, *, msg):
...
+
         else:
-            self._set_failed('unknown', 'Action failed with unknown reason')
-            logger.error('Received unknown action result status %s', msg)
+            # Got some other result figure out if it is a result that indicates the action can be retried
```
Comment (MarkWesleyAtAnki) [inline src/cozmo/action.py]: We try and wrap at about 80 chars for the SDK, suggest rewording to:
```
            # All other results should fall under either the abort or retry
            # categories, determine the category by shifting the result.
```

File: src/cozmo/action.py [hunk 3]
```diff
@@ -219,9 +210,21 @@ def _recv_msg_robot_completed_action(self, evt, *, msg):
...
+            resultCategory = result >> _clad_to_game_cozmo.ARCBitShift.NUM_BITS
+            resultCategories = _clad_to_game_cozmo.ActionResultCategory
+            
+            if resultCategory == resultCategories.ABORT:
+                self._set_failed('aborted', 'Action failed')
```
Comment (MarkWesleyAtAnki) [inline src/cozmo/action.py]: Ideally we would add the result to the message, so e.g. 'Action failed: NO_FACE', but we don't have an automatic enum->string conversion in the python clad generator currently, so you have to build something each time - we can add this later though, so don't worry about it for now (this is mostly just a note for myself).

Comment (alchaussee) [inline src/cozmo/action.py]: Yeah we should definitely do that
